### DISCUSSION TITLE
[docs][guides] improve Schema Customization docs

### DISCUSSION THREAD

**Discussion by marcysutton** [gatsbyjs/gatsby#31213] [Ideas / Feature Requests]
## Summary

Through feedback, we've heard that the [Schema Customization docs](https://www.gatsbyjs.org/docs/schema-customization/) could use some improvement for understanding how to apply the concepts. For example:

> "Gives too specific of examples without really explaining how the concepts work. Having a difficult time applying schema customization to my own use case. (want to make a new connection based on MarkdownRemarkConnection)"

> "it makes a bit of sense, but i really want to see it in action, and i'm not sure I can get a type builder working end to end in my project"

The doc uses an example project, which is great. But it would be helpful to add some more ideas throughout the doc to help users understand how to apply the concepts to their own work (e.g. "One use case for this approach would be ______", "You might apply this to do ____", etc.)

This doc also needs to be updated to adhere to the [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun), as it uses "our" and "we" instead of focusing on the reader with "you" and "your".

### Motivation

We want users of all skill levels to feel more comfortable using Schema Customization. It's a powerful API that we aren't quite explaining in a way that people can understand -- therefore its usefulness is limited to more advanced users. In general, adding examples of _why_ and _how_ to use something are just as important as the explanation of _what_ the thing is -- we've seen a similar situation with Gatsby Themes.

### Related issues

Loosely related to #14228, which is more focused on Gatsby Internals.

**Comment by DaniTulp** [comment]
I can work on the Gatsby Style guide rules. I can try to add some more examples, but I am currently not experienced (yet) with all the ins and outs of the Schema Customization API.

**Comment by marcysutton** [comment]
Here are some materials that include a wealth of relevant information, particularly the context on "Why it was needed" and themes use cases:

- https://www.gatsbyjs.org/blog/2019-03-04-new-schema-customization/ 
- https://www.christopherbiscardi.com/post/constructing-query-types-in-themes/

**Comment by caitlinselca** [comment]
Hi! I am a new contributor to Gatsby and this particular issue interests me a lot. May I begin working on this ticket to resolve the issue?

**Comment by kzuri** [comment]
Hello @marcysutton, I would like to contribute to this issue
